### PR TITLE
Add support for Client Credentials for MicrosoftGraphApplications

### PR DESCRIPTION
**Pull request by ostolero**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3696)

## Description
Adding support for authenticating using Client Credentials.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_description.md [hunk 1]
```diff
@@ -17,5 +26,18 @@ In order to use the Cortex XSOAR Azure application, use the default application
 
 To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal, with mobile and desktop flows enabled.
 
+### Client Credentials Flow
+___
+Follow these steps for a self-deployed configuration:
+
+1. To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal. To add the registration, refer to the following [Microsoft article](https://docs.microsoft.com/en-us/microsoft-365/security/defender/api-create-app-web?view=o365-worldwide#create-an-app) steps 1-8.
```
Comment (JudahSchwartz) [inline Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_description.md:33]: looks like a repeat of a few lines up.

File: Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/README.md [hunk 2]
```diff
@@ -1,17 +1,21 @@
+Use the Azure Active Directory Applications integration to manage authorized applications.
 Use the Azure Active Directory Applications integration to manage authorized application.
 This integration was integrated and tested with version v1.0 of Microsoft Graph Services API.
-## Configure MicrosoftGraphServices on Cortex XSOAR
+## Configure Azure Active Directory Applications on Cortex XSOAR
```
Comment (JudahSchwartz) [inline Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/README.md:4]: was this on purpose?

File: Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/README.md [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
-    | **Parameter** | **Required** |
-    | --- | --- |
-    | Application ID | True |
-    | Azure AD endpoint | False |
-    | Trust any certificate (not secure) | False |
-    | Use system proxy settings | False |
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | Application ID |  | True |
+    | Azure AD endpoint | Azure AD endpoint associated with a national cloud. | False |
```
Comment (JudahSchwartz) [inline Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/README.md:13]: Azure AD endpoint?

**Review by JudahSchwartz — COMMENTED** [review]
looks great.
Looks like you accidentally copied from AD and forgot to switch the name back

**Review by JudahSchwartz — APPROVED** [review]

File: Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_description.md [hunk 4]
```diff
@@ -17,5 +26,18 @@ In order to use the Cortex XSOAR Azure application, use the default application
 
 To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal, with mobile and desktop flows enabled.
 
+### Client Credentials Flow
```
Comment (yucohen) [inline Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_description.md:29]: Lets put this pare above the device code, as this is the recommended authentication method

Comment (yucohen) [inline Packs/MicrosoftGraphApplications/Integrations/MicrosoftGraphApplications/MicrosoftGraphApplications_description.md:29]:
```suggestion
### Client Credentials Flow (Recommended)
```

**Review by yucohen — COMMENTED** [review]
Great!

**Pull request merged**